kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -240,6 +254,14 @@ public Optional<FileStatus> getLastSeenChecksum() {
     return lastSeenChecksum;
   }
 
+  /**
+   * Returns the maximum version among all published delta files that were seen during log segment
+   * construction, if available.
+   */
+  public Optional<Long> getMaxPublishedDeltaVersion() {
```

Comment:
Can you elaborate on why it might be empty? Does that mean nothing has been published, or some other case where we just didn't see it (due to listing bounds or something)?

Reply:
Great question.

1. Yup I'm leaving the door open for a table of only catalog commits with nothing published.
2. Yup looking at our LogSegment construction code in SnapshotManager .. it's easy to get lost in the sauce of guarantees and invariants and what not. 
3. We are adding this so catalog clients can inform their server of the max published version -- no need for it to be non-optional actually

For those 3 reasons I was inclined to just move fast and make it optional. Simple and clear API, simple implementation. And now the caller of this method can decide what it wants to do if it is optional.empty.

Reply:
I guess:

4. Perhaps we listed and ONLY found a checkpoint file (due to log cleanup) And perhaps we update kernel java to allow this (Snapshot::getTimestamp would fail for an ICT table). Thus, unknown max published delta version

Reply:
Can you add some minor version of this context to the docs (sorry I was not clear, that's what I meant in my first comment). Maybe just gentle guidance on how a caller should interpret/treat an empty optional?

Reply:
Follow up question, is it possible this is Optional.of(-1) if no versions have been published? Or is it empty in that case?